### ISSUE TITLE
ssh_sftp:write_file_info/3,4 returns {error,failure}

### ISSUE THREAD

**Issue by shapovalovts** [erlang/otp#7895]
**Describe the bug**
`ssh_sftp:write_file_info` constantly returns `{error,failure}`.

**To Reproduce**
1. Generate host keys in /tmp/ssh
2. Run this script that starts ssh daemon locally and connect to it, performing operations:

``` erlang
#!/usr/bin/env escript                                                                                                                                    
                                                                                                                                                          
-include_lib("kernel/include/file.hrl").                                                                                                                  
                                                                                                                                                          
-define(SSH_PORT, 31000).                                                                                                                                 
                                                                                                                                                          
main(_) ->                                                                                                                                                
    {ok, _} = application:ensure_all_started(ssh),                                                                                                        
                                                                                                                                                          
    SshDaemonOpts = [                                                                                                                                     
        {system_dir, "/tmp/ssh"},                                                                                                       
        {password, "test"}                                                                                                                                
    ],                                                                                                                                                    
    {ok, _} = ssh:daemon(?SSH_PORT, SshDaemonOpts),                                                                                                       
                                                                                                                                                          
    ConnOpts = [                                                                                                                                          
        {password, "test"},                                                                                                                               
        {silently_accept_hosts, true},                                                                                                                    
        {preferred_algorithms, ssh:default_algorithms()}                                                                                                  
    ],                                                                                                                                                    
    {ok, ConnectionRef} = ssh:connect("localhost", ?SSH_PORT, ConnOpts),                                                                                  
    {ok, Pid} = ssh_sftp:start_channel(ConnectionRef),                                                                                                    
                                                                                                                                                          
    ok = ssh_sftp:write_file(Pid, "/tmp/file", <<"Test">>),                                                                                               
    {ok, FileInfo} = file:read_file_info("/tmp/file"),                                                                                                    
    Result = ssh_sftp:write_file_info(Pid, "/tmp/file", FileInfo#file_info{mode = 8#00777}),                                                              
                                                                                                                                                          
    io:format("File info: ~p~n", [FileInfo]),                                                                                                             
    io:format("Result: ~p~n", [Result]).        
``` 
The script prints:
```
File info: {file_info,4,regular,read_write,
                      {{2023,11,21},{18,19,45}},
                      {{2023,11,21},{18,19,45}},
                      {{2023,11,21},{18,19,45}},
                      33188,1,53,0,12978347,1000,1000}
Result: {error,failure}
```

**Expected behavior**
`ssh_sftp:write_file_info` either works fine ( which is expected in the script above) or prints meaningful error message. But never just "failure".

**Affected versions**
 Reproduced on both OTP 24.2 and 26.1.2.

**Comment by u3s** [comment]
I think code above will work if `FileInfo` is removed from line
`Result = ssh_sftp:write_file_info(Pid, "/tmp/file", FileInfo#file_info{mode = 8#00777}), `

and simply `#file_info{mode = 8#00777}` is used as 3rd argument. this is verified in `ssh_sftp_SUITE:set_attributes/1`

I guess some of the fields of in file_info record can't be set and cause `write_file_info` failure.
I agree error message is not perfect. Plan to investigate more and get back.

**Comment by u3s** [comment]
I think root cause is the same as #7897.
Both issues will be fixed by #8220.

**Comment by u3s** [comment]
fix is merged to maint and master branches. planned to be released in upcoming patches for OTP-26, OTP-25 and OTP-24.
